### PR TITLE
PM-5955 Multiple Badges Overlap With Permissions Column

### PR DESCRIPTION
**Pull request by Jingo88**
```
- [ X ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Update the badge component so it can be given a width that will not overlap with the column next to it, in this case the permissions column. Also updated the badges so the first two items will not be truncated. 

## Code changes

* `vault-collection-row` Added a viewchild to get the width of the Groups column to pass to the badge components
* `group-name-badge` Updated to take width input and pass it down to `badge-list`
* `badge-list` updated to apply a width to style if given. Updated to not truncate all badges, just the last one or a long one

## Screenshots

Updated Screenshot to show wrapping badges when they exceed parent width

Screenshot 2024-02-08 at 4 43 15 PM

### CODE REVIEW

**Review by LRNcardozoWDF — DISMISSED** [review]
LGTM!

**Review by willmartian — CHANGES_REQUESTED** [review]
Generally, I'm not a fan on passing in a specific pixel width to components. Pixels aren't responsive to the font size changing. I think a cleaner solution would be to have the badge list respect the width of its parent (in this case, the table cell). 

Also, I'm not thrilled with the way these convey info:
image

I would prefer that the 1st example displays as: [Admin] [+3 more]
3rd example: [Admin] [+2 more]

However, that change is a bit more complicated and would require inspecting the parent width with JS. We could also flex-wrap badges. 

@bitwarden/dept-design, what is the desired look/behavior for truncated badge lists?

**Comment by Jingo88** [comment]
Adding another option for design. if we don't need everything inline it would look like this

Screenshot 2024-02-08 at 2 17 24 PM

**Comment by sukhleenb** [comment]
Definitely prefer your suggestion of wrapping the badges Will!

**Review by sukhleenb — APPROVED** [review]

**Review by LRNcardozoWDF — DISMISSED** [review]

File: libs/components/src/badge-list/badge-list.component.html [hunk 1]
```diff
@@ -1,8 +1,13 @@
-<div class="tw-inline-flex tw-gap-2">
+<div [ngClass]="{ 'tw-inline-flex': variant !== 'secondary', 'tw-gap-2': true }">
```
Comment (willmartian) [inline libs/components/src/badge-list/badge-list.component.html:1]: I don't think we want to couple the color variant with the display mode. 

Also, this should still be a flex container to properly utilize `gap`. I think we just need to turn on [flex wrap](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-wrap) with `tw-flex-wrap`.

File: libs/components/src/badge-list/badge-list.component.html [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
   <ng-container *ngFor="let item of filteredItems; let last = last">
-    <span bitBadge [variant]="variant" [truncate]="truncate">
+    <span bitBadge *ngIf="!last; else lastBadge" [variant]="variant" [truncate]="false">
       {{ item }}
     </span>
+    <ng-template #lastBadge>
+      <span bitBadge [variant]="variant" [truncate]="truncate">
+        {{ item }}
+      </span>
+    </ng-template>
```
Comment (willmartian) [inline libs/components/src/badge-list/badge-list.component.html:10]: I'm not sure if these changes are warranted. What if the first item in a badge list is very long? We would still want it to truncate, no?

**Review by willmartian — CHANGES_REQUESTED** [review]

File: libs/components/src/badge-list/badge-list.component.html (hunk 1 above)
Comment (Jingo88) [reply libs/components/src/badge-list/badge-list.component.html]: awesome thanks!

**Review by willmartian — APPROVED** [review]
LGTM! Thanks for iterating with me @Jingo88

**Review by LRNcardozoWDF — APPROVED** [review]

**Pull request merged**